pkg/stub/handler.go

```diff
@@ -460,23 +506,27 @@ func (h *Handler) processFiles(dir string, files []os.FileInfo, opcfg *v1alpha1.
 
 func (h *Handler) updateDockerPullSpec(oldies []string, imagestream *imagev1.ImageStream, opcfg *v1alpha1.SamplesResource) {
 	if len(opcfg.Spec.SamplesRegistry) > 0 {
-		replaced := false
-		for _, old := range oldies {
-			if strings.HasPrefix(imagestream.Spec.DockerImageRepository, old) {
-				imagestream.Spec.DockerImageRepository = strings.Replace(imagestream.Spec.DockerImageRepository, old, opcfg.Spec.SamplesRegistry, 1)
-				replaced = true
-				break
+		if !strings.HasPrefix(imagestream.Spec.DockerImageRepository, opcfg.Spec.SamplesRegistry) {
+			// if not one of our 4 defaults ...
+			replaced := false
+			for _, old := range oldies {
+				if strings.HasPrefix(imagestream.Spec.DockerImageRepository, old) {
+					imagestream.Spec.DockerImageRepository = strings.Replace(imagestream.Spec.DockerImageRepository, old, opcfg.Spec.SamplesRegistry, 1)
+					replaced = true
+					break
+				}
+			}
+			if !replaced {
+				imagestream.Spec.DockerImageRepository = opcfg.Spec.SamplesRegistry + "/" + imagestream.Spec.DockerImageRepository
 			}
-		}
-		if !replaced {
-			opcfg.Spec.SamplesRegistry = opcfg.Spec.SamplesRegistry + "/" + imagestream.Spec.DockerImageRepository
 		}
 
 		for _, tagref := range imagestream.Spec.Tags {
 			replaced := false
 			for _, old := range oldies {
 				if tagref.From != nil && tagref.From.Kind == "DockerImage" {
-					if strings.HasPrefix(tagref.From.Name, old) {
+					if !strings.HasPrefix(tagref.From.Name, opcfg.Spec.SamplesRegistry) &&
```

Comment:
why not move this outside the "oldies" loop like you did for DockerImageRepository?

Reply:
Just missed that ... thanks, part of next push